tests/function/test_config.py

```diff
@@ -29,50 +28,122 @@ def test_debug_in_virtwho_conf(self, virtwho, globalconf):
+        """Test the interval option in /etc/virtwho.conf
 
         :title: virt-who: config: test interval option
         :id: f1d39429-62c0-44f0-a6d3-4ffc8dc704b1
         :caseimportance: High
-        :tags: tier2
+        :tags: tier1
         :customerscenario: false
         :upstream: no
         :steps:
-            1.
+
+            1. Enable interval and set to 10 in /etc/virt-who.conf
+            2. Enable interval and set to 60 in /etc/virt-who.conf
+            3. Enable interval and set to 120 in /etc/virt-who.conf
         :expectedresults:
-            1.
+
+            1. Default value of 3600 seconds will be used when configure lower than 60 seconds
+            2. Configure successfully, and virt-who starting infinite loop with 60 seconds interval
+            3. Configure successfully, and virt-who starting infinite loop with 120 seconds interval
         """
-        logger.info("Succeeded to run the 'test_interval_in_virtwho_conf'")
+        globalconf.update('global', 'debug', 'True')
+        globalconf.update('global', 'interval', '10')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['interval'] == 3600)
+
+        globalconf.update('global', 'interval', '60')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['interval'] == 60)
 
-    @pytest.mark.tier3
-    def test_oneshot_in_virtwho_conf(self):
-        """Just a demo
+        globalconf.update('global', 'interval', '120')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['interval'] == 120)
+
+    @pytest.mark.tier1
+    def test_oneshot_in_virtwho_conf(self, virtwho, globalconf):
+        """Test the oneshot option in /etc/virtwho.conf
 
         :title: virt-who: config: test oneshot option
-        :id: 15b45691-7358-4bc8-b49f-f87100bded1b
+        :id: 9e39f91f-80b5-4773-bef0-7facf8cb85e2
         :caseimportance: High
-        :tags: tier3
+        :tags: tier1
         :customerscenario: false
         :upstream: no
         :steps:
-            1.
+
+            1. Run virt-who with "oneshot=True" in /etc/virt-who.conf
+            2. Run virt-who with "oneshot=False" in /etc/virt-who.conf file
+
         :expectedresults:
-            1.
+
+            1. Can see 'Thread X stopped after running once' log in rhsm.log
+            2. Cannot see 'Thread X stopped after running once' log in rhsm.log
         """
-        logger.info("Succeeded to run the 'test_oneshot'")
+        globalconf.update('global', 'debug', 'True')
+        globalconf.update('global', 'oneshot', 'True')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['oneshot'] is True)
```

Comment:
Please add checking the `result['terminate']` when test the oneshot, the ['oneshot'] is also just to check the keyword in log, the terminate is to make sure the virt-who serivce is really killed.

Reply:
Added